### ISSUE TITLE
Minecraft version matching will not match against prereleases

### ISSUE THREAD

**Issue by OpenBagTwo** [OpenBagTwo/EnderChest#55]
## Summary

A `shulkerbox.cfg` with the following section:

```ini
[minecraft]
*
```

will not match to an instance with minecraft version of, say,

```ini
minecraft_version = 1.20-pre6
```

## Environment and Version
- EnderChest version: enderchest v0.1.0-rc2 
- rsync version: N/A 
- Python version: 3.11.3 
- Operating System: Linux x86_64 
- `semantic_version` version: 2.10.0

## Steps to Reproduce

1. Create an instance in PrismLauncher using the latest pre-release
1. Register that instance in your enderchest, _e.g._
   ```bash
   $ enderchest gather minecraft instances/snapshot
   ```
1. Create a shulker box with default specs (name it, say, "global")
1. Run the command:
   ```bash
   $ enderchest inventory shulker_box global
   ```

and the pre-release instance will not appear in the list of matches

## Workaround

Remove the `[minecraft]` section or manually add the specific version

## Severity

Gamebreaking

## Desired Outcome

Fix

## Urgency

Needs to be part of the current release

## Notes

Ultimately this is an issue with my use of `semantic_version`. I say "my use of" because it might not even technically be wrong.

The simple solution is to build in special handling for "*", but I have to decide whether, for example, `1.20-pre6 >= 1.19.4` should evaluate to `True`.

**Comment by OpenBagTwo** [comment]
The *ideal* solution would be to load and parse `version_manifest_v2.json` and use _that_ to determine whether a version fell inside a range. And it's not even like I'd need to download or maintain it myself--it should be findable on the user's system already (and we could fall back to semver if it's not there). But that's some really nasty complexity.

This might be a matter of:

1. Quick fix now
1. "Correct" fix later.

**Comment by OpenBagTwo** [comment]
semver, you were unfairly maligned. It turns out I swapped my `fnmatch` argument order. Again. :man_facepalming:

**Comment by OpenBagTwo** [comment]
There was also an issue with Minecraft not using proper semver, so it would be really hard to match a shulker box to, say, _just_ 1.19 instances, but that's been fixed in da4b479cc5f4c9579db5edfffdd826dc2ad79d03
